cds_migrator_kit/videos/weblecture_migration/load/helpers.py

```diff
@@ -0,0 +1,211 @@
+        options = frames_task._time_position(
+            duration=object_version.get_tags()["duration"],
+            frames_start=0,  # TODO they should be changed with using the frame file name?
+            frames_end=90,
+            frames_gap=10,
+        )
+        frames = frames_task._create_frames(
+            frame_paths,
+            object_version,
+            options.get("start_time"),
+            options.get("time_step"),
+        )
+        output_folder = tempfile.mkdtemp()
+        frames_task._create_gif(
+            bucket=str(object_version.bucket.id),
+            frames=frames,
+            output_dir=output_folder,
+            master_id=version_id,
+        )
+        flow_task_metadata.status = FlowTaskStatus.SUCCESS
+
+        # Cleanup
+        shutil.rmtree(output_folder)
+    except Exception:
+        shutil.rmtree(output_folder, ignore_errors=True)
+        frames_task.clean(version_id=version_id)
+        raise ManualImportRequired("Frame uploading failed!", stage="load")
+
+
+def load_subformats(payload, subformat_paths):
+    """Load subformats for the master video file without celery task.
+
+    payload (dict):
+        - version_id (str): The version ID of the main video file.
+        - key (str): The file name of the main video file.
+        - bucket_id (str): The bucket id of the record.
+        - deposit_id (str): The deposit id of the record.
+        - flow_id (str): The flow id.
+    subformat_paths (list): A list of file paths of the subformats.
+    """
+    # TODO should we need to check all the subformats exists? If missing should we generate?
+    # TODO maybe logging is enough?
+
+    # Create FlowTaskMetadata
+    subformat_payload = payload.copy()
+    transcode_task = FlowTaskMetadata.create(
+        flow_id=payload["flow_id"],
+        name=TranscodeVideoTask.name,
+        payload=subformat_payload,
+    )
+
+    transcode_task.status = FlowTaskStatus.STARTED
+
+    for path in subformat_paths:
+        video_name = os.path.basename(path)
+        preset_quality = None
+
+        # TODO this regex might be changed for the composite subformats
+        # Get quality from file name
+        match = re.search(r"-(\d{3,4}p)-quality\.[a-zA-Z0-9]+$", video_name)
+        if match:
+            preset_quality = match.group(1)  # Extracts the quality part (e.g. "480p")
+        if not preset_quality:
+            raise ManualImportRequired(f"Subformat quality not found.", stage="load")
+
+        # Copy the file to FileInstance and create ObjectVersion
+        obj = load_file_to_bucket(payload["bucket_id"], path)
+
+        # Add tags to the subformat
+        ObjectVersionTag.create(obj, "master", payload["version_id"])
+        ObjectVersionTag.create(obj, "media_type", "video")
+        ObjectVersionTag.create(obj, "context_type", "subformat")
+        ObjectVersionTag.create(obj, "smil", "true")
+        ObjectVersionTag.create(obj, "_opencast_event_id", "")  # TODO is it needed?
```

Comment:
Hmmm, I think you could omit it, the only consequence I can think of is some logged errors triggered by the celery task that is checking for started tasks on opencast to update them.